pkg/status/status.go

```diff
@@ -151,3 +151,16 @@ func MessageFromError(err error) string {
 	}
 	return unwErr.Error()
 }
+
+func resetIncomparableConditionFields(conditions []metav1.Condition) {
+	for idx := range conditions {
+		conditions[idx].LastTransitionTime = metav1.Time{}
+		conditions[idx].ObservedGeneration = 0
+	}
+}
```

Comment:
are we guaranteed this mutates the given objects?

Reply:
yes because `conditions[idx]` will modify the underlying array element at index `idx` whereas if this was a var defined in the `for` statement (or for scoped) and is updated later then the array wouldn't be mutated because we updated a copy:
https://go.dev/play/p/TZwmvA6_nJH

Reply:
thanks for writing an example on go playground! even better would have been point out to a testcase in our codebase.